client/web/src/search/input/SearchContextCtaPrompt.tsx

```diff
@@ -38,14 +40,27 @@ export const SearchContextCtaPrompt: React.FunctionComponent<SearchContextCtaPro
             : `/users/${authenticatedUser.username}/settings/code-hosts`
         : '/sign-up'
 
+    const onClick = useCallback(() => {
+        const actionKind = authenticatedUser
+            ? hasUserAddedExternalServices
+                ? 'AddRepositories'
+                : 'ConnectCodeHost'
+            : 'SignUp'
```

Comment:
nit: would it be easier to read without nested ternary expression?
We don't have an ESLint rule for that yet, but we will consider it in the near future.

```suggestion
        const authenticatedActionKind = hasUserAddedExternalServices ? 'AddRepositories' : 'ConnectCodeHost'
        const actionKind = authenticatedUser ? authenticatedActionKind : 'SignUp'
```